jdaviz/configs/imviz/plugins/reproject/reproject.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+import time
+from glue.core.data import Data
+from traitlets import Bool, Unicode
+
+from jdaviz.core.events import SnackbarMessage
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin, DatasetSelectMixin
+
+try:
+    from reproject import reproject_interp
+    from reproject.mosaicking import find_optimal_celestial_wcs
+except ImportError:
+    HAS_REPROJECT = False
+else:
+    HAS_REPROJECT = True
+
+__all__ = ['Reproject']
+
+
+@tray_registry('imviz-reproject', label="Reproject")
+class Reproject(PluginTemplateMixin, DatasetSelectMixin):
+    template_file = __file__, "reproject.vue"
+
+    results_label = Unicode("Reprojected").tag(sync=True)
+    reproject_in_progress = Bool(False).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not HAS_REPROJECT:
+            self.disabled_msg = 'Please install reproject and restart Jdaviz to use this plugin'
+            return
+
+    def vue_do_reproject(self, *args, **kwargs):
+        if (not HAS_REPROJECT or self.dataset_selected not in self.data_collection
+                or self.reproject_in_progress):
+            return
+
+        import reproject
+
+        data = self.data_collection[self.dataset_selected]
+        wcs_in = data.coords
+        if wcs_in is None:
+            return
+
+        viewer_reference = f"{self.app.config}-0"
+        self.reproject_in_progress = True
+        t_start = time.time()
+        try:
+            if self.results_label in self.app.data_collection.labels:
+                raise ValueError(
+                    f'{self.results_label} is already used, choose another label name.')
+
+            # Find WCS where North is pointing up.
+            wcs_out, shape_out = find_optimal_celestial_wcs([(data.shape, wcs_in)], frame='icrs')
+
+            # Reproject image to new WCS.
+            comp = data.get_component(data.main_components[0])
+            new_arr = reproject_interp((comp.data, wcs_in), wcs_out, shape_out=shape_out,
+                                       return_footprint=False)
```

Comment:
Sorry if this has already been discussed and I'm repeating a question. Should we support reprojection onto non-optimal WCS? For example, one might want to rotate one image so that it matches the orientation of another (see [reproject Quick Start](https://reproject.readthedocs.io/en/stable/index.html#quick-start)), which might be helpful for blinking in Imviz.

Reply:
Ori wanted N-up, so that is what he is getting. Arbitrary rotation would be a different ticket.